kbs/src/api_server.rs

```diff
@@ -0,0 +1,317 @@
+
+        let http_config = self.config.http_server.clone();
+        let http_server = HttpServer::new({
+            move || {
+                let api_server = self.clone();
+                App::new()
+                    .wrap(middleware::Logger::default())
+                    .app_data(web::Data::new(api_server))
+                    .service(
+                        web::resource([kbs_path!("{plugin}{sub_path:.*}")])
+                            .route(web::get().to(client))
+                            .route(web::post().to(client)),
+                    )
+                    .service(
+                        web::resource([kbs_path!("admin/{plugin}/{sub_path:.*}")])
+                            .route(web::get().to(admin))
+                            .route(web::post().to(admin)),
+                    )
+            }
+        });
+
+        if !http_config.insecure_http {
+            let tls_server = http_server
+                .bind_openssl(
+                    &http_config.sockets[..],
+                    crate::http::tls_config(&http_config)
+                        .map_err(|e| Error::HTTPSFailed { source: e })?,
+                )
+                .map_err(|e| Error::HTTPSFailed { source: e.into() })?;
+
+            return tls_server
+                .run()
+                .await
+                .map_err(|e| Error::ServerLaunchFailed { source: e.into() });
+        }
+
+        http_server
+            .bind(&http_config.sockets[..])
+            .map_err(|e| Error::HTTPFailed { source: e.into() })?
+            .run()
+            .await
+            .map_err(|e| Error::ServerLaunchFailed { source: e.into() })
+    }
+}
+
+/// Client APIs. /kbs/v0/XXX
+pub(crate) async fn client(
+    request: HttpRequest,
+    body: web::Bytes,
+    core: web::Data<ApiServer>,
+) -> Result<HttpResponse> {
+    let query = request.query_string();
+    let plugin_name = request
+        .match_info()
+        .get("plugin")
+        .ok_or(Error::IllegalAccessedPath {
+            path: request.path().to_string(),
+        })?;
+    let sub_path = request
+        .match_info()
+        .get("sub_path")
+        .ok_or(Error::IllegalAccessedPath {
+            path: request.path().to_string(),
+        })?;
+
+    let end_point = format!("{plugin_name}{sub_path}");
+
+    match plugin_name {
+        #[cfg(feature = "as")]
+        "auth" if request.method() == Method::POST => core
+            .attestation_service
+            .auth(&body)
+            .await
+            .map_err(From::from),
+        #[cfg(feature = "as")]
+        "attest" if request.method() == Method::POST => core
+            .attestation_service
+            .attest(&body, request)
+            .await
+            .map_err(From::from),
+        #[cfg(feature = "as")]
+        "attestation-policy" if request.method() == Method::POST => {
+            core.admin_auth.validate_auth(&request)?;
+
+            core.attestation_service.set_policy(&body).await?;
```

Comment:
This might be out of scope of this PR, but there is an open question regarding the security of this endpoint when we are not using a built-in AS. Basically the AS doesn't really evaluate where these requests are coming from.

Reply:
The api makes less sense when we use non-built-AS and we usually directly interact with AS to set the policy. The existing of this API is much like a workaround for all-in-one-kbs, where the admin should care about the policy. I personally turn to use separate AS and KBS rather than all-in-one-kbs.

Some more thoughts. There is a nice architecture from classic [Kerberos](https://en.wikipedia.org/wiki/Kerberos_(protocol)). It decouples AS (like CoCoAS in our case, to verify the valiadity of evidence), TGS (like the authorization part of KBS in our case) and SS (like the resource storage + policy engine of KBS in our case). This would have a clear barrier of each components and helps maintain each of them without mixing.